### ISSUE TITLE
When using TypeScript, the types related to the .process() function are not correct

### ISSUE THREAD

**Issue by kevintechie** [supercharge/promise-pool#27]
The types of the .process function are not being handled correctly causing compile errors:

```TypeScript
import Pool from '@supercharge/promise-pool';

export default class MyClass {
  myConcurrency: number;

  constructor(concurrency: number) {
    this.myConcurrency = concurrency;
  }

  public async processStuff(stuff: Array<number>): Promise<number[]> {
    const pool = new Pool()
      .withConcurrency(this.myConcurrency)
      .for(stuff);

    const { results } = await pool.process((x: number) => x + 1);

    return results;
  }
}
```

The TypeScript compiler gives the following error:

```text
Argument of type '(x: number) => number' is not assignable to parameter of type '(item: unknown) => number | Promise<number>'.
  Types of parameters 'x' and 'item' are incompatible.
    Type 'unknown' is not assignable to type 'number'.

15     const { results } = await pool.process((x: number) => x + 1);
                                              ~~~~~~~~~~~~~~~~~~~~
```

**Comment by marcuspoehls** [comment]
@kevintechie Hey Kevin, thank you for reporting this issue. Do you want to send a PR fixing this?

**Comment by marcuspoehls** [comment]
Fixed in `1.6.1`.

I just published version `1.6.1` on NPM. Should be available in a second 😃
